### PR TITLE
Create SPEC_ASSUMPTIONS.md

### PR DESCRIPTION
**Pull request by OisinNolan**

### CODE REVIEW

File: SPEC_ASSUMPTIONS.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+<div aria-labelledby='foo'></div>
+```
+Does div#foo’s aria-labelledby=’bar’ count as being part of an aria-labelledby traversal? Chrome says no, and for this example acc-name of div#foo = ‘Some words and some more words’.
+
+We assume that a node must be directly referenced by aria-labelledby to be part of an aria-labelledby traversal.
+#### Related GitHub Issues
+- https://github.com/w3c/accname/issues/79
+
+## C.1
+We assume that context.partOfName is true implies that the current node is being traversed due to recursion. We can make this assumption because partOfName is set to true any time the algorithm is called recursively.
```
Comment (AlexLloyd0) [inline SPEC_ASSUMPTIONS.md:55]: Is this a spec assumption?

File: SPEC_ASSUMPTIONS.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+## D.1
+We assume that the [HTML Accessibility API Mappings spec](https://www.w3.org/TR/html-aam-1.0/#accessible-name-and-description-computation) lists all instances in which native elements or attributes define a text alternative. It is this document that we used as a guide to implement 2D.
+## E.1
```
Comment (AlexLloyd0) [inline SPEC_ASSUMPTIONS.md:58]: The assumptions for `2E` don't have any references. Do no issues on the spec exist for thesse? Should they?

File: SPEC_ASSUMPTIONS.md (hunk 1 above)
Comment (OisinNolan) [reply SPEC_ASSUMPTIONS.md:55]: No actually, this is more of a meta-spec-assumption. I'm thinking that simply a comment in the code might be the most appropriate way to document this assumption as it's not directly related to the spec?

Comment (AlexLloyd0) [reply SPEC_ASSUMPTIONS.md:55]: Sounds good. It's less clear but  you could also consider this for the `CON` assumptions

File: SPEC_ASSUMPTIONS.md (hunk 2 above)
Comment (OisinNolan) [reply SPEC_ASSUMPTIONS.md:62]: Added links to my own contributions in new commit

File: SPEC_ASSUMPTIONS.md (hunk 1 above)
Comment (OisinNolan) [reply SPEC_ASSUMPTIONS.md:55]: I think the CON assumptions are technically spec assumptions, right? Although the 'context object' is our own abstraction, it is entirely based of assumptions that different conditions in the spec are equivalent, i.e. the context object itself is kind of like a big spec assumption. What do you think?

**Review by AlexLloyd0 — APPROVED** [review]

**Pull request merged**